src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioErrorReportingServiceFactory.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Extensions;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation
+{
+    [ExportWorkspaceServiceFactory(typeof(IErrorReportingService), ServiceLayer.Host), Shared]
+    internal sealed class VisualStudioErrorReportingServiceFactory : IWorkspaceServiceFactory
+    {
+        private readonly VisualStudioWorkspaceImpl _workspace;
+        private IErrorReportingService _singleton;
+
+        [ImportingConstructor]
+        public VisualStudioErrorReportingServiceFactory(VisualStudioWorkspaceImpl workspace)
+        {
+            _workspace = workspace;
+        }
+
+        public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
+        {
+            return _singleton ?? (_singleton = new VisualStudioErrorReportingService(_workspace));
```

Comment:
if it is singleton for host only why use factory?

any reason not adding one for other layer even if it doesn't do anything? IErrorReportingService sounds like generic enough?